prog/mutation.go

```diff
@@ -148,17 +149,128 @@ func (ctx *mutator) removeCall() bool {
 	return true
 }
 
+// Generates a random argument index based on prorities associated with each argument.
+func randomChoice(m map[float64][]int, r *randGen) (int){
```

Comment:
In Go code it is more common to put more important things first, and then less important things afterwards.
You follow C style where a code reader is hit with the least important things first because of declaration requirements. Go does not require forward declarations for things.
Please revert order of functions. Generally if foo calls bar, then bar is located after foo.